docs/security-considerations.rst

```diff
@@ -273,17 +290,17 @@ control to a fixed and trusted third party or just converts the contract into
 a simple "give me back my money" contract.
 
 
-*******************
-Formal Verification
-*******************
+Method visibility from mother contracts and ABI
+===============================================
+
+
+When compiling a contract that inherits from other contracts, the generated JSON ABI for the child contract does not contain all the methods from contracts in the upper hierarchy. However, that does not prevent public methods from being executed. You should take care of which methods can be executed. Use `internal` function modifier to control visibility.
```

Comment:
I think the Parity bug is a very special case which we could mention: do not `delegatecall` into a library from a fallback, because that exposes any public method of the library.

The problem is not having public methods in a library, but proxying the entire incoming message without any checks.